templates/todo/api/java/pom.xml

```diff
@@ -74,6 +74,11 @@
       <artifactId>jackson-databind</artifactId>
     </dependency>
     <!-- Azure dependencies -->
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
+      <version>1.9.0-beta.1</version>
+    </dependency>
```

Comment:
How is the dependency used? No changes to credentials being used anywhere?

Reply:
we use `boot spring` for the java template (which I really don't like hahha)

As a dependency we use: `<artifactId>spring-cloud-azure-starter-keyvault-secrets</artifactId>` which is configured from `resources/application.properties`  with:

```
spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
```
This configuration makes `boot spring` to use the `DefaultAzureCredential` to connect to Key Vault and get all the secrets from as `application-properties`.  The application then can access the secret keys as if they were EnvVars.

So, later we set:
`spring.data.mongodb.uri=${AZURE_COSMOS_CONNECTION_STRING:#{null}}`

where `AZURE_COSMOS_CONNECTION_STRING` is the key used in Key Vault for the secret.  So, when the application runs, the secret is ready for mongo to connect :), with the magic of `spring-cloud-azure-starter` 

I had to break all of this before to introduce the `AzureDeveloperCredential` locally by disabling the `spring-cloud-azure-starter` as:

```
# Temporally disabling auto-configuration for key vault
# Import KeyVault secrets as properties
# spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
# spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
custom.azure.keyvault.secret.mongodb=${AZURE_KEY_VAULT_ENDPOINT},AZURE-COSMOS-CONNECTION-STRING
```
Then, I created a `bean` to manually connect to Key Vault to pull `AZURE-COSMOS-CONNECTION-STRING` secret and create the sdk client.  

Now I just remove that hack and let the auto-config to work again.. as the AzureDeveloperCredential is not part of the DefaultAzureCredential on Identity sdk lib :)